Add support for EDL Token authentication to the /s3credentials endpoint

yjpa7145 opened the pull request:
If the `/s3credentials` endpoint of the distribution API receives a request with the `EDL-ClientId` and `EDL-Token` headers set, it will authenticate the request using that token rather than using the OAuth process.

Jkovarik commented on tf-modules/distribution/index.js:
```diff
@@ -177,6 +178,32 @@ function isPublicRequest(path) {
+  req.get('EDL-Client-Id') && req.get('EDL-Token');
+
+const handleTokenAuthRequest = async (req, res, next) => {
+  const earthdataLoginClient = EarthdataLogin.createFromEnv({
```
This duplicates a few lines from getConfigurations() - it might be nice to DRY that into a small helper, use getConfigurations or some other refactor here.

Jkovarik commented:
Approving, conditionally assuming the tokens/user query has been tested with an application group.   Nice work!

yjpa7145 commented:
Manual testing I did for this s3credentials stuff:

- Worked with URS to create an application group
- Created a “mth-cumulus-dev” app
- Created a “mth-cumulus-ci” app
- Added those two apps to the group
- Updated the /token endpoint to print out the actual EDL token it gets to Cloudwatch Logs
- Configured my stack to use the “-dev” app
- Deployed my stack
- Hit the API’s /token endpoint
- Looked in the Cloudwatch Logs to get the EDL token that was created
- Redeployed my stack using the “-ci” app credentials
- Hit the s3credentials endpoint with the “EDL-Client-Id” and “EDL-Token” headers set
- Confirmed that I got AWS credentials back
